mstranslator.py

```diff
@@ -89,143 +94,109 @@ def make_response(self, resp):
+            {'text' : text} for text in texts
+        ]
+        return self._translate('translate', body, lang_from, lang_to,
+                               contenttype, category, includeAlignment = True)
 
     def break_sentences(self, text, lang):
         if len(text) > 10000:
             raise ValueError('The text maximum length is 10000 characters')
         params = {
-            'text': text,
             'language': lang,
         }
-        lengths = self.make_request('BreakSentences', params)
+        body = [
+            {'Text': text}
+        ]
+        lengths = self.make_request('breaksentence', params, body)
+        print lengths
         if isinstance(text, bytes):
             text = text.decode('utf-8')
         c = 0
         result = []
-        for i in lengths:
+        for i in lengths[0]['sentLen']:
             result.append(text[c:c+i])
             c += i
         return result
 
-    def add_translation(self, text_orig, text_trans, lang_from, lang_to, user, rating=1,
-                        contenttype='text/plain', category='general', url=None):
-        if len(text_orig) > 1000:
-            raise ValueError('The original text maximum length is 1000 characters')
-        if len(text_trans) > 2000:
-            raise ValueError('The translated text maximum length is 1000 characters')
-        if contenttype not in ('text/plain', 'text/html'):
-            raise ValueError('Invalid contenttype value')
-        if not -10 < rating < 10 or not isinstance(rating, int):
-            raise ValueError('Raiting must be an integer value between -10 and 10')
+    def get_langs(self):
         params = {
-            'originalText': text_orig,
-            'translatedText': text_trans,
-            'from': lang_from,
-            'to': lang_to,
-            'user': user,
-            'contentType': contenttype,
-            'rating': rating,
-            'category': category,
+            'scope': 'translation'
         }
-        if url:
-            params['uri'] = url
-        return self.make_request('AddTranslation', params)
 
-    def get_langs(self, speakable=False):
-        action = 'GetLanguagesForSpeak' if speakable else 'GetLanguagesForTranslate'
-        return self.make_request(action)
+        response = self.make_request('languages', params, isPost = False)
+        result = []
+        for lang in response['translation']:
+            result.append(lang)
+        return result
 
     def get_lang_names(self, langs, lang_to):
         params = {
-            'locale': lang_to,
-            'languageCodes': json.dumps(langs),
+            'scope': 'translation'
+        }
+        headers = {
+            'Accept-Language': lang_to
         }
-        return self.make_request('GetLanguageNames', params)
+        response = self.make_request('languages', params, headers = headers, isPost = False)
+        result = []
+        for lang in langs:
+            if(lang in response['translation']):
```

Comment:
No braces around if case is required. `if lang in response['translation']:`. And list-comprehensions would also be nice here.

Reply:
I'm not used to use python, I have changed.